mock/mockdb/vbucket.go

```diff
@@ -161,7 +168,10 @@ func (s *Vbucket) findDocLocked(repIdx, collectionID uint, key []byte) *Document
 func (s *Vbucket) pushDocMutationLocked(doc *Document) *Document {
 	newDoc := copyDocument(doc)
 	newDoc.VbUUID = s.currentUUIDLocked()
-	newDoc.SeqNo = s.nextSeqNoLocked()
+	newDoc.SeqNo = doc.SeqNo
+	if newDoc.SeqNo == 0 {
```

Comment:
I am pretty concerned about doing this since we read/modify/write a lot, which means we could accidentally reuse an existing seqno by mistake.  Also, this causes the seqno's to end up in our mockdb in the incorrect order.